2010-10-20  Yao Qi  <yao@codesourcery.com>

	Merge from Sourcery G++ to fix LP:660021
	2010-10-18  Paul Brook  <paul@codesourcery.com>

	* tree-vect-stmts.c (supportable_widening_operation): Check if wide
	vector type exists.

=== modified file 'gcc/tree-vect-stmts.c'
--- a/gcc/tree-vect-stmts.c	2010-07-23 15:51:08 +0000
+++ b/gcc/tree-vect-stmts.c	2010-10-20 16:52:04 +0000
@@ -4650,6 +4650,11 @@
   tree wide_vectype = get_vectype_for_scalar_type (type);
   enum tree_code c1, c2;
 
+  /* Check we have a valid vector type for the result.  */
+  if (!wide_vectype)
+    return false;
+
+
   /* The result of a vectorized widening operation usually requires two vectors
      (because the widened results do not fit int one vector). The generated
      vector results would normally be expected to be generated in the same

